spark/src/main/java/org/apache/zeppelin/spark/SparkRInterpreter.java

```diff
@@ -0,0 +1,218 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.spark;
+
+import static org.apache.zeppelin.spark.ZeppelinRDisplay.render;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.zeppelin.interpreter.*;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * R and SparkR interpreter with visualization support.
+ */
+public class SparkRInterpreter extends Interpreter {
+  private static final Logger logger = LoggerFactory.getLogger(SparkRInterpreter.class);
+
+  private static String renderOptions;
+
+  static {
+    Interpreter.register(
+      "r",
+      "spark",
+      SparkRInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+              .add("spark.master",
+                      SparkInterpreter.getSystemDefault("MASTER", "spark.master", "local[*]"),
+                      "Spark master uri. ex) spark://masterhost:7077")
+              .add("spark.home",
+                      SparkInterpreter.getSystemDefault("SPARK_HOME", "spark.home", "/opt/spark"),
+                      "Spark distribution location")
+              .add("zeppelin.R.image.width",
+                      SparkInterpreter.getSystemDefault("ZEPPELIN_R_IMAGE_WIDTH",
+                              "zeppelin.R.image.width", "100%"),
+                      "")
+              .add("zeppelin.R.render.options",
+                      SparkInterpreter.getSystemDefault("ZEPPELIN_R_RENDER_OPTIONS",
+                              "zeppelin.R.render.options",
+                              "out.format = 'html', comment = NA, "
+                                      + "echo = FALSE, results = 'asis', message = F, warning = F"),
+                      "")
+              .build());
+  }
+
+  public SparkRInterpreter(Properties property) {
+    super(property);
+  }
+
+  @Override
+  public void open() {
+    zeppelinR().open(getProperty("spark.master"),
+            "/opt/spark", getSparkInterpreter());
```

Comment:
Should this use spark.home?
in conf/zeppelin-env.sh if I set 
export SPARK_HOME=/someotherpath/spark

The interpreter still tries to use /opt/spark

I had to create a symlink from /someotherpath/spark to /opt/spark to get it to work.